packages/core/src/database/index.ts

```diff
@@ -1167,45 +1183,156 @@ $$ LANGUAGE plpgsql
       await this.wrap(
         { method: "removeTriggers", includeTraceLogs: true },
         async () => {
-          for (const tableName of getTableNames(schemaBuild.schema)) {
-            await sql
-              .raw(
-                `DROP TRIGGER IF EXISTS "${tableName.trigger}" ON "${namespace}"."${tableName.sql}"`,
-              )
-              .execute(qb.internal);
+          for (const table of tables) {
+            await qb.drizzle.execute(
+              sql.raw(
+                `DROP TRIGGER IF EXISTS "${getTableNames(table).trigger}" ON "${namespace}"."${getTableName(table)}"`,
+              ),
+            );
           }
         },
       );
     },
-    async revert({ checkpoint }) {
+    getStatus() {
+      return this.wrap({ method: "_ponder_status.get()" }, async () => {
+        const result = await this.qb.drizzle.select().from(PONDER_STATUS);
+
+        if (result.length === 0) {
+          return null;
+        }
+
+        const status: Status = {};
+
+        for (const row of result) {
+          status[row.network_name] = {
+            block:
+              row.block_number && row.block_timestamp
+                ? {
+                    number: row.block_number,
+                    timestamp: row.block_timestamp,
+                  }
+                : null,
+            ready: row.ready,
+          };
+        }
+
+        return status;
+      });
+    },
+    setStatus(status) {
+      return this.wrap({ method: "_ponder_status.set()" }, async () => {
+        await this.qb.drizzle
+          .insert(PONDER_STATUS)
+          .values(
+            Object.entries(status).map(([networkName, value]) => ({
+              network_name: networkName,
+              block_number: value.block?.number,
+              block_timestamp: value.block?.timestamp,
+              ready: value.ready,
+            })),
+          )
+          .onConflictDoUpdate({
+            target: PONDER_STATUS.network_name,
+            set: {
+              block_number: sql`excluded.block_number`,
+              block_timestamp: sql`excluded.block_timestamp`,
+              ready: sql`excluded.ready`,
+            },
+          });
+      });
+    },
+    async revert({ checkpoint, db }) {
```

Comment:
The `db`/`tx` polymorphism is weird in cases like this, because it's definitely not safe to run this without using a transaction.